simulator/src/main/java/simblock/node/Node.java

```diff
@@ -315,8 +325,13 @@ public void addOrphans(Block orphanBlock, Block validBlock) {
   public void minting() {
     AbstractMintingTask task = Simulator.getConsensusAlgo().CreateMintingTask(this);
     this.mintingTask = task;
+
     if (task != null) {
```

Comment:
Just because a node has a winning hash, does not mean it gets to mint the block. The node most have a winning hash and it must be the nodes turn. A node is marked ready when it has a winning hash.